src/tabs/chat/ChatTabController.tsx

```diff
@@ -0,0 +1,69 @@
+import { TabController } from "../TabController";
+import { ChatTabState, Message } from "./ChatTabState";
+import { ChatAutoComplete } from "./ChatAutocomplete";
+
+export class ChatTabController extends TabController<ChatTabState> {
+    public toggleCommandView = () => {
+        var oldState = this.model.getState();
+        this.model.update({
+            commandMode: !oldState.commandMode
+        })
+    }
+
+    public execute = () => {
+        var command: string = this.model.getState().commandToSend;
+        if (this.model.getState().commandMode) {
+            if (this.IsChatCommand(command)) {
+                this.model.requestResource({
+                    data: command,
+                    method: "POST",
+                    resourcePath: ["console"],
+                });
+
+                this.model.update({ commandToSend: "", messageSendStatus: "SENDING" });
+            }
+            else {
+                let oldModel = this.model.getState();
+                let errorMessage: any = {
+                    "type": "CLIENT",
+                    "message": "Invalid chat command. Only `say` and `whisper`.",
+                    "newLine": true
+                };
+                this.model.update({
+                    messages: oldModel.messages.concat(errorMessage as Message),
+                });
+            }
+        }
+        else{
+            let m = this.model.getState();
+            console.log(m);
+            let recipient: String = m.selectedRecipient;
+            if(!recipient){
+                recipient = "ALL"
+            }
+            if(recipient == "ALL"){
```

Comment:
I think it would be fine if you moved the contents of this if statement into the one above.